packages/core/contracts/financial-templates/perpetual-multiparty/PerpetualCreator.sol

```diff
@@ -65,7 +69,28 @@ contract PerpetualCreator is ContractCreator, Testable, Lockable {
      * @return address of the deployed contract.
      */
     function createPerpetual(Params memory params) public nonReentrant() returns (address) {
-        address derivative = PerpetualLib.deploy(_convertParams(params));
+        // Create a new synthetic token using the params.
```

Comment:
This creator creates the Synthetic token and checks that permissions are set correctly. It then passes the newly created synthetic token to the Perpetual